Sources/ProjectDescription/HeaderFileGlob.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  HeaderFileGlob.swift
+//  ProjectDescription
+//
+//  Created by Paweł Trafimuk on 01/12/2021.
+//
+
+import Foundation
+
+/// A type that represents a list of source files defined by a glob.
+public struct HeaderFileGlob: Codable, Equatable {
+    /// Glob pattern to header files
+    public var glob: Path
+
+    /// Relative glob patterns for excluded files.
+    public var excluding: [Path]
+
+    /// Initializes the header file glob.
+    /// - Parameters:
+    ///   - glob: Glob pattern to header files
+    ///   - excluding: Glob pattern used for filtering out files.
+    public init(_ glob: Path,
```

Comment:
I'd make this one private and expose a static method called `glob` that makes the resulting code more idiomatic:

```swift
.glob("headers/**", excluding: [])

vs

HeaderFileGlob("headers/**", excluding: [])
.init("headers/**", excluding: [])
```

Reply:
This is how I imagine the API being:

```swift
.headers(public: [.glob("headers/public/**")])
```

Reply:
done